datafusion/datasource-parquet/src/source.rs

```diff
@@ -720,7 +721,9 @@ impl FileSource for ParquetSource {
         let filters: Vec<PushedDownPredicate> = filters
             .into_iter()
             .map(|filter| {
-                if can_expr_be_pushed_down_with_schemas(&filter, &file_schema) {
+                if is_dynamic_physical_expr(&filter)
+                    || can_expr_be_pushed_down_with_schemas(&filter, &file_schema)
```

Comment:
Shouldn't we update `can_expr_be_pushed_down_with_schemas` to accept partition values or `can_expr_be_pushed_down_with_schemas(&filter, &table_schema)`?

Reply:
So I had included it earlier, but I removed it later with this mental model:

- This function receives filters directly mentioned in query and dynamic filters both.
- For a filter found in a query, if it contains a partition column, it would be partition pruned by table provider.
- The only case where this function would receive a filter expr containing partition column, is for dynamic filter
- That means, we want to lower dynamic filter regardless of any case, so we add a check if expr is dynamic expr, we let it lower down

Do correct me if I am misunderstanding anything here :)

Reply:
> For a filter found in a query, if it contains a partition column, it would be partition pruned by table provider.
> The only case where this function would receive a filter expr containing partition column, is for dynamic filter

I don't think this is right. Consider `file_data_column < partition_column` as a filter. That can only be pruned once we have the file data, despite referring to a partition column. This actually used to be a bug in DataFusion, fixed in https://github.com/apache/datafusion/pull/15057 I believe. Now `ParquetOpener` can evaluate filters that reference partition columns.

Reply:
checking failed tests